libpromises/locks.c

```diff
@@ -709,20 +709,23 @@ CfLock AcquireLock(EvalContext *ctx, const char *operand, const char *host, time
     time_t lastcompleted = FindLock(cflast);
     time_t elapsedtime = (time_t) (now - lastcompleted) / 60;
 
-    if (elapsedtime < 0)
+    if (tc.ifelapsed != 0)
```

Comment:
Maybe a short comment in the lines of "In case of ifelapsed==0 we don't skip the promise in case a parallel agent re-executes it at the same time."